app/controllers/games_controller.rb

```diff
@@ -15,6 +16,18 @@ def index
     @games = Game.all
   end
 
+  def update
+    @game = Game.find_by_id(params[:id])
+    return render_not_found unless @game.present?
+    return render_not_found(:unauthorized) if @game.white_player_id.present? && @game.black_player_id.present?
+    if @game.white_player_id.present?
+      @game.update_attributes(black_player_id: current_user.id)
+    else
+      @game.update_attributes(white_player_id: current_user.id)
+    end
+    redirect_to root_path
+  end
```

Comment:
For any method, another trick you can do for improving readability.

Try to group/separate similar logic with new lines. It makes it easier for people to quickly read. So for example.

```Ruby
def somemethod?
  return if something # guard statements/checking params

  something = find_something
  something = find_somethingelse # loading data

  if something # logic of the method
    do_something
  else something_else
     do_other_thing
  end
end
```

Notice how I group similar logic together / separate it with new lines. Kind of like paragraphs when you're writing.

There is an awesome blog post/chapter in a book on this, I'll try to find it. They explain better than I do. :)